app/src/main/java/com/droidhats/campuscompass/viewmodels/MapViewModel.kt

```diff
@@ -1,7 +1,23 @@
 package com.droidhats.campuscompass.viewmodels
 
+import android.app.Activity
 import androidx.lifecycle.ViewModel
+import com.droidhats.campuscompass.models.Campus
+import com.droidhats.campuscompass.repositories.MapRepository
+import java.io.InputStream
 
 class MapViewModel : ViewModel() {
-    // TODO: Implement the ViewModel
+
+    private var campuses: List<Campus>? = null
+
+    fun getCampuses(): List<Campus> = campuses!!
+
+    // The activity is required to access the assets to open our json file where the info
+    // is stored
+    fun init(activity: Activity) {
```

Comment:
I have come to realise that passing the activity context to a view model could cause memory leaks due to them having different lifecycles. If the class really needs the context it should extend AndroidViewModel instead.